mu/interface.py

```diff
@@ -865,17 +877,22 @@ def dropEvent(self, event):
         source = event.source()
         self.disable(source)
         if isinstance(source, LocalFileList):
-            local_filename = os.path.join(self.home,
-                                          source.currentItem().text())
-            logger.info("Putting {}".format(local_filename))
-            try:
-                with microfs.get_serial() as serial:
-                    logger.info(serial.port)
-                    microfs.put(serial, local_filename)
-                super().dropEvent(event)
-            except Exception as ex:
-                logger.error(ex)
+            file_exists = self.findItems(source.currentItem().text(),
+                                         Qt.MatchExactly)
+            if not file_exists or \
+                    file_exists and self.show_confirm_overwrite_dialog():
+                local_filename = os.path.join(self.home,
+                                              source.currentItem().text())
+                logger.info("Putting {}".format(local_filename))
+                try:
+                    with microfs.get_serial() as serial:
+                        logger.info(serial.port)
+                        microfs.put(serial, local_filename)
+                    super().dropEvent(event)
+                except Exception as ex:
+                    logger.error(ex)
         self.enable(source)
+        self.parent().ls()
```

Comment:
Yeah, I think this area probably deserves some consideration for future refactor, as the interface probably shouldn't be doing any of the microfs logic to `ls()` populate the `MicrobitFileList` list. And since `ls` is a method from `FileSystemPane`, now `LocalFileList` and `MicrobitFileList` would depend on being created inside a PyQt widget with an `ls()` method.

For the moment we could have a mandatory argument for `parent` in `MuFileList` (like `FileSystemPane` on any PyQt widget really) and then mock that for the tests. Seems a bit cleaner than checking if it has a parent here, although nothing stops a developer from creating the instance with a `parent=None` argument and having this crash here anyway, just like they could have an inherent parent without the `ls()` method and it would crash here as well.